cedar-lean/Cedar/Thm/Validation.lean

```diff
@@ -0,0 +1,70 @@
+/-
+ Copyright 2022-2023 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+      https://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+-/
+
+import Cedar.Spec
+import Cedar.Data
+import Cedar.Validation
+import Cedar.Thm.Validation.Validator
+
+/-!
+This file contains the top-level correctness properties for the Cedar validator.
+We show that if validation succeeds, then for any request consistent with the schema,
```

Comment:
Suggested rephrasing:

```
Suggested rephrasing:  

```
We show that if validation succeeds for a set of policies, then for any request consistent with the schema,
evaluating a policy in this set either produces a boolean value, or throws one of the errors in the set of valid errors.
```
```